[Feature] 회원 정보 변경, 비밀번호 변경, 내 프로필 조회 기능 구현

moomint8 opened the pull request:
### 🧑‍💻PR 타입(하나 이상의 PR 타입을 선택해주세요)🧑‍💻
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [x] 코드 및 구조 개선

### 반영 브랜치🥞
ex) feat/auth -> dev

### 변경 사항🧐
- 회원 정보 변경 기능 구현
- 비밀번호 변경 기능 구현
- 회원 프로필 조회 기능 구현
- JWT Filter 개선

### 테스트 결과🔖
![image](https://github.com/user-attachments/assets/186f1a74-53b7-449d-8941-5731589ece44)

Leegiyeon reviewed (approved):
프롬프트 내용 수정 확인되었습니다.
좋은 결과물을 위힌 지속적인 수정 감사드려요 👍

moomint8 committed:
test(User): 회원 정보 변경 테스트 구현

회원 정보를 변경하는 테스트를 구현했습니다.

moomint8 committed:
feat(User): 회원 정보 변경 기능 구현

회원 정보를 변경하는 기능을 구현했습니다.
회원 정보는 성별과 휴대폰 번호 변경만 가능합니다.
선정 기준은 둘 모두 null able 이므로 첫 가입시 미입력하는 경우를 상정했기 때문입니다.
비밀번호는 이전, 이후 모두 확인해야하는 특성상 따로 분리할 예정입니다.
그 외의 정보는 변경되지 않을 것이라 판단되어 선정하지 않았습니다.

moomint8 committed:
feat(User): 회원 정보 변경 Controller 구현

회원 정보를 변경하는 Controller를 구현했습니다.

moomint8 committed:
refactor(User): JWT Filter 개선

잘못된 토큰이 들어오는 경우를 처리하고, 예상치 못한 예외에 대해서도 우선 처리 후 개선이 가능하도록 로깅 처리를 추가했습니다.

moomint8 committed:
test(User): 비밀번호 변경 테스트 구현

비밀번호 변경 기능 테스트를 구현했습니다.
성공 케이스와 이전 비밀번호가 틀린 예외 케이스를 테스트합니다.

moomint8 committed:
feat(User): 비밀번호 변경 기능 구현

비밀번호 변경 기능을 구현했습니다.

moomint8 committed:
feat(User): 비밀번호 변경 Controller 구현

비밀번호 변경 Controller을 구현했습니다.

moomint8 committed:
feat(User): 내 프로필 조회 Controller 구현

내 프로필 조회 Controller를 구현했습니다.

moomint8 committed:
Merge branch 'dev' of https://github.com/AI-Jiwoo/Jiwoo-Back into feat/user

Leegiyeon commented on src/main/java/org/jiwoo/back/common/filter/JwtFilter.java:
```diff
@@ -31,51 +37,67 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
         if (authorization == null || !authorization.startsWith("Bearer ")) {
 
-            log.info("Access without token: HTTP Method = {}, URI = {}", request.getMethod(), request.getRequestURI());
+            log.info("Access without token: HTTP Method = {}, URI = {}", httpMethod, requestUri);
```
이게 어제말한 로그군여🤗

Leegiyeon reviewed (approved):
회원 정보 수정 기능 확인했습니다~ 🤓 
고생하셨습니다아 :)
